File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260917-060211-45590/analysis/own-tests-project/src/main/java/customerimporter/CUIT.java
package customerimporter;

import java.util.List;
import java.util.function.Supplier;

public class CUIT extends Identification {

    private static final String TYPE = "C";
    private static final List<String> VALID_HEADERS = List.of("20", "23", "24", "25", "26", "27", "30", "33", "34");

    // instance creation

    public static boolean canHandle(String aType) {
        return TYPE.equals(aType);
    }

    // error messages

    public static String invalidCUITNumberErrorDescription() {
        return "Invalid CUIT number";
    }

    // initialization

    public CUIT(String aNumber) {
        super(aNumber);
    }

    // accessing

    public String type() {
        return TYPE;
    }

    // testing

    public boolean isDNI() {
        return false;
    }

    public boolean isCUIT() {
        return true;
    }

    public Object dniNumberIfNone(Supplier<Object> aNoneBlock) {
        return aNoneBlock.get();
    }

    public Object cuitNumberIfNone(Supplier<Object> aNoneBlock) {
        return number;
    }

    // validation

    // "23-25666777-9" size 13
    protected void assertIsValid() {
        int penultimate = number.length() - 2;

        assertThat(number.length() >= 12 && number.length() <= 13);
        assertThat(number.charAt(2) == '-' && number.charAt(penultimate) == '-');
        assertThat(VALID_HEADERS.contains(number.substring(0, 2)));
        assertThat(Character.isDigit(number.charAt(number.length() - 1)));
        assertThat(Characters.areAllDigits(number.substring(3, penultimate)));
    }

    protected String invalidNumberErrorDescription() {
        return invalidCUITNumberErrorDescription();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-213435-24827/analysis/acceptance-project/src/customerimporter/cuit.py
from customerimporter.identification import Identification


class Cuit(Identification):

    # instance creation

    @classmethod
    def type_code(cls):
        return "C"

    @classmethod
    def valid_headers(cls):
        return ["20", "23", "24", "25", "26", "27", "30", "33", "34"]

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid CUIT number"

    # validation

    def assert_is_valid(self):
        # "23-25666777-9" size 13
        self._assert_that(len(self._value) >= 12 and len(self._value) <= 13)
        self._assert_that(self._value[2] == "-" and self._value[-2] == "-")
        self._assert_that(self._value[:2] in type(self).valid_headers())
        self._assert_that(self._value[-1].isdigit())
        self._assert_that(self._are_all_digits(self._value[3:-2]))

    # cuit

    def cuit_number(self):
        return self._value

    def cuit_number_if_none(self, a_none_closure):
        return self.cuit_number()

    def has_cuit_as_identification(self):
        return True
